### PR TITLE
Enable checking stack allocation before converting to LLVM dialect.

### PR DESCRIPTION
**Pull request by hanhanW**
The flag is not removed because sometimes we might want to bypass the check.

### CODE REVIEW

**Comment by hanhanW** [comment]
According to the [discussion](https://discord.com/channels/689900678990135345/689906000043573354/967105460396261436), I'll update the check to verify if the total stack allocation is less than 16 KB.

File: iree/compiler/Codegen/LLVMCPU/LLVMCPUCheckIRBeforeLLVMConversion.cpp [hunk 1]
```diff
@@ -22,12 +22,25 @@ struct LLVMCPUCheckIRBeforeLLVMConversionPass
...
   auto moduleOp = getOperation();
   // For now only check that there are no stack allocations.
-  auto walkResult = moduleOp.walk([](memref::AllocaOp allocaOp) -> WalkResult {
-    return allocaOp.emitOpError("expected no static allocations");
+  int64_t bits = 0;
+  auto walkResult = moduleOp.walk([&](memref::AllocaOp allocaOp) -> WalkResult {
+    auto type = allocaOp.getType().cast<ShapedType>();
+    if (!type.hasStaticShape()) {
+      return allocaOp.emitOpError(
+          "expected no static allocations with dynamic shapes");
```
Comment (pzread) [inline iree/compiler/Codegen/LLVMCPU/LLVMCPUCheckIRBeforeLLVMConversion.cpp:30]: nit: stack allocation?

File: iree/compiler/Codegen/LLVMCPU/LLVMCPUCheckIRBeforeLLVMConversion.cpp [hunk 2]
```diff
@@ -22,12 +22,25 @@ struct LLVMCPUCheckIRBeforeLLVMConversionPass
...
+    bits += type.getSizeInBits();
+    return WalkResult::advance();
   });
   if (walkResult.wasInterrupted()) {
     return signalPassFailure();
   }
+  constexpr int k16KBInBits = 16 * 1024 * 8;
+  if (bits >= k16KBInBits) {
+    moduleOp.emitOpError(
+        "expected total size of static allocation is smaller than 16 KB");
```
Comment (pzread) [inline iree/compiler/Codegen/LLVMCPU/LLVMCPUCheckIRBeforeLLVMConversion.cpp:41]: nit: Same here

**Review by pzread — APPROVED** [review]

File: iree/compiler/Codegen/LLVMCPU/LLVMCPUCheckIRBeforeLLVMConversion.cpp [hunk 3]
```diff
@@ -22,12 +22,25 @@ struct LLVMCPUCheckIRBeforeLLVMConversionPass
 void LLVMCPUCheckIRBeforeLLVMConversionPass::runOnOperation() {
   auto moduleOp = getOperation();
   // For now only check that there are no stack allocations.
```
Comment (pzread) [inline iree/compiler/Codegen/LLVMCPU/LLVMCPUCheckIRBeforeLLVMConversion.cpp:24]: nit: Remove this comment?

**Pull request merged**